assignments/prototypes.js

```diff
@@ -32,7 +51,58 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
- 
+function Humanoid(humanoidAttributes){
+  // GameObject.call(this, humanoidAttributes);
+  CharacterStats.call(this, humanoidAttributes);
+  this.faction = humanoidAttributes.faction;
+  this.weapons = humanoidAttributes.weapons;
+  this.language = humanoidAttributes.language;
+}
+
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+Humanoid.prototype.greet = function(){
+  return `${this.name} offers a greeting in ${this.language}.`;
+}
+
+
+/*
+  === Hero ===
+*/
+function Hero(heroAttributes){
+  Humanoid.call(this, heroAttributes);
+}
+
+Hero.prototype = Object.create(Humanoid.prototype);
+
+Hero.prototype.heroAttack = function(victim){
+  victim.hp = victim.hp - 2;
+  
+  if(victim.hp > 0) {
+    return `${this.name} heroically beat up ${victim.name}. ${victim.takeDamage()} -2 hp.
+  ${victim.name}: ${victim.hp} remaining`;
+  } else return victim.destroy();
+}
+
+/*
+  === Villain ===
+*/
+function Villain(villainAttributes){
+  Humanoid.call(this, villainAttributes);
+}
+
+Villain.prototype = Object.create(Humanoid.prototype);
+
+Villain.prototype.villainAttack = function(victim){
+  victim.hp = victim.hp - 14;
+  
+  if(victim.hp > 0) {
+    return `${this.name} snuck up behind ${victim.name} and stabbed him in the back! ${victim.takeDamage()} -14 hp
+  ${victim.name}: ${victim.hp} remaining`;
+  } else return victim.destroy();
```

Comment:
the keyword `else` is not needed here as you are `return`ing from inside the `if` statement